low-level/framework/rabbitmq/rabbitmq_egress_processor.py

```diff
@@ -276,10 +276,13 @@ def _transmit_msg_on_exchange(self):
                 self._add_signature()
                 jsonMsg = json.dumps(self._jsonMsg).encode('utf8')
                 try:
-                    self._connection.publish(exchange=self._exchange_name,
-                                            routing_key=self._routing_key,
-                                            properties=msg_props,
-                                            body=jsonMsg)
+                    if self.store_queue.is_empty():
+                        self._connection.publish(exchange=self._exchange_name,
+                                                routing_key=self._routing_key,
+                                                properties=msg_props,
+                                                body=jsonMsg)
+                    else:
```

Comment:
good. add a log.info() also here stating the action, with reason. keep it short

Reply:
added.